Source/TaskDelegate.swift

```diff
@@ -33,7 +33,7 @@ public class TaskDelegate: NSObject {
     /// The serial operation queue used to execute all operations after the task completes.
     public let queue: OperationQueue
 
-    let task: URLSessionTask
+    var task: URLSessionTask
```

Comment:
This is the most important part of this change. By allowing the underlying `task` to be swapped on the `TaskDelegate`, we can essentially defer the suspended queue operations while swapping out the `task` inside the same `Request`. This will allow us to create a new task from the `Request` and swap the task in the `TaskDelegate`.